.buildkite/scripts/steps/serverless-integration-tests.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+# This script runs the serverless integration tests
+set -eo pipefail
+
+source .buildkite/scripts/common2.sh
+
+# Make sure that all tools are installed
+asdf install
+
+echo "~~~ Building test binaries"
+mage build:testBinaries
+
+# TODO: move to common.sh when it's refactored
+# BK analytics
+echo "--- Prepare BK test analytics token :vault:"
+BUILDKITE_ANALYTICS_TOKEN=$(vault kv get -field token kv/ci-shared/platform-ingest/buildkite_analytics_token)
+export BUILDKITE_ANALYTICS_TOKEN
```

Comment:
This env variable should not be here IMO, but in the `pre-command` hook, that will avoid situations where `BUILDKITE_ANALYTICS_TOKEN` can be printed in plain in the logs.

Or use https://github.com/elastic/vault-secrets-buildkite-plugin if we want to be more explicit about where that env variable needs to be accessed.

Reply:
See https://github.com/elastic/elastic-agent/blob/f1086184d87ed755b9275081c3b42f58c54feb63/.buildkite/hooks/pre-command#L49-L66


IIUC, `if [[ "$BUILDKITE_STEP_KEY" == *"integration-tests"* ]]; then` will enable that.


However, I don't see we are using the BK Test plugin:
- https://github.com/buildkite-plugins/test-collector-buildkite-plugin

Maybe it could be deleted? Or enable the `key: something-integration-tests-something` so the `pre-command` hook can honour it?